### PR TITLE
Remove incorrect or useless dependencies

### PR DESCRIPTION
**Pull request by loheagn**
1.  `github.com/docker/docker` from `v1.6.1` to `v20.10.17+incompatible` ( to be consistent with [nerdctl](https://github.com/containerd/nerdctl/blob/5f291d236399dbf913ea90cf91b36b5c4c6a6501/go.mod#L24))
2. remove `github.com/docker/libcontainer`

Signed-off-by: Nan Li <loheagn@icloud.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -37,10 +37,9 @@ require (
 	github.com/cpuguy83/go-md2man/v2 v2.0.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/docker/distribution v2.8.1+incompatible // indirect
-	github.com/docker/docker v1.6.1 // indirect
+	github.com/docker/docker v1.4.2-0.20190924003213-a8608b5b67c7 // indirect
```
Comment (AkihiroSuda) [inline go.mod:40]: Why not use v20.10.xx

Comment (loheagn) [reply go.mod:40]: Because this version is imported by `github.com/containerd/containerd v1.6.6`

**Review by AkihiroSuda — APPROVED** [review]

**Comment by hsiangkao** [comment]
may I ask what we can do for this pr? it'd be much better to have nerdctl support ;)

**Review by imeoer — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (loheagn) [reply go.mod]: Hi, @AkihiroSuda , I have upgraded `github.com/docker/docker` to `v20.10.17+incompatible` :)

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**